src/number_format.js

```diff
@@ -360,22 +375,34 @@ class NumberFormat extends React.Component {
   /** caret specific methods ends **/
 
 
-  /** methods to remove formattting **/
+  /** methods to remove formatting **/
   removePrefixAndSuffix(val: string) {
-    const {format, prefix, suffix} = this.props;
+    const {format, prefix, suffix, negationFormat} = this.props;
 
     //remove prefix and suffix
     if (!format && val) {
-      const isNegative = val[0] === '-';
+      let isNegative = val[0] === '-';
+      let stringEnd = val.length;
+
+      if (negationFormat === 'parentheses') {
+        isNegative = val[0] === '(' && val[val.length - 1] === ')';
+        if (isNegative) {
+          stringEnd = stringEnd - 1;
+        }
+      }
 
       //remove negation sign
-      if (isNegative) val = val.substring(1, val.length);
+      if (isNegative) val = val.substring(1, stringEnd);
 
       //remove prefix
       val = prefix && val.indexOf(prefix) === 0 ? val.substring(prefix.length, val.length) : val;
 
       //remove suffix
-      const suffixLastIndex = val.lastIndexOf(suffix);
+      let suffixLastIndex = val.lastIndexOf(suffix);
+      if (negationFormat === 'parentheses' && isNegative) {
+        suffixLastIndex = suffixLastIndex - 1;
```

Comment:
This logic looks incorrect, as you have already removed the parenthesis. I don't think we need to do suffixLastIndex - 1 here. Moreover, even though you wouldn't have removed parenthesis, lastIndexOf(suffix) will always give the correct value.